src/main.rs

```diff
@@ -301,7 +307,20 @@ async fn forward_request(
         })
         .collect::<Vec<_>>();
 
-    let entire_body = body::to_bytes(body).await?.to_vec();
+    // Limit request body size
+    let body = Limited::new(body, REQUEST_BODY_SIZE_LIMIT);
+    let entire_body = match hyper::body::to_bytes(body).await {
```

Comment:
If you wanted to rename `entire_body` to `body` (because `body` is already eaten by this code so it wouldn't even be shadowing), I wouldn't hate it.